### PR TITLE
read prefetch list(runtime) from file

### PR DESCRIPTION
**Pull request by Desiki-high**
change the nydusd  arg --prefetch as a file path, we can read the prefetch list from the file. For example:
we can use `--prefetch-files /tmp/test` and the `/tmp/test` is like 
```
/etc
/bin
/src/test.py
```
When nydusd is mounted, this file is read by line to form a prefetch list

### CODE REVIEW

File: src/bin/nydusd/main.rs [hunk 1]
```diff
@@ -220,10 +220,10 @@ fn append_fs_options(app: Command) -> Command {
     .arg(
         Arg::new("prefetch-files")
             .long("prefetch-files")
-            .help("List of files/directories to prefetch")
+            .help("A file include the list of files/directories to prefetch")
```
Comment (changweige) [inline src/bin/nydusd/main.rs:223]: "A file including xxx"

**Review by changweige — COMMENTED** [review]
Is it a breaking change? Please also update the test cases in nydus-test.

File: api/openapi/nydus-api-v1.yaml [hunk 2]
```diff
@@ -348,10 +348,8 @@ components:
           description: usually to be the metadata source
           type: string
         prefetch_files:
-          description: files that need to be prefetched
-          type: array
-          items:
-            type: string
+          description: local file path which recorded a file list to be prefetched
```
Comment (bergwolf) [inline api/openapi/nydus-api-v1.yaml:351]: Better be explicit that the list of file paths is separated with newlines.

**Review by jiangliu — APPROVED** [review]

**Comment by jiangliu** [comment]
It's a break change, any comments from @changweige ?

**Comment by jiangliu** [comment]
It may be worth to set a upper limit on prefetch-list-file?

**Comment by changweige** [comment]
> It's a breaking change, any comments from @changweige ?

Though I am not a big fan of changing the semantics of nydusd arguments, it still looks good. Hopefully, there is no user enormously depending on the old semantics

File: contrib/nydus-test/functional-test/test_nydus.py [hunk 3]
```diff
@@ -564,7 +564,7 @@ def test_pseudo_fs(nydus_anchor, nydus_image: RafsImage, rafs_conf: RafsConf):
     nc.umount_rafs("/pseudo3")
 
 
-def test_shared_blobcache(nydus_anchor: NydusAnchor, nydus_image, rafs_conf: RafsConf):
+def test_shared_blobcache(nydus_anchor: NydusAnchor, nydus_image, rafs_conf: RafsConf, tmp_path):
```
Comment (changweige) [inline contrib/nydus-test/functional-test/test_nydus.py:567]: We can't add a parameter to the test function if it is not a fixture. Did you successfully run the test?

File: contrib/nydus-test/functional-test/test_nydus.py [hunk 4]
```diff
@@ -573,13 +573,16 @@ def test_shared_blobcache(nydus_anchor: NydusAnchor, nydus_image, rafs_conf: Raf
...
+
     def make_rafs(mountpoint):
         rafs = (
             NydusDaemon(nydus_anchor, nydus_image, rafs_conf)
             .apisock(tempfile.NamedTemporaryFile().name)
-            .prefetch_files("/")
+            .prefetch_files(os.path.abspath(tmp_file))
             .set_mountpoint(mountpoint)
-        )
+        )
```
Comment (changweige) [inline contrib/nydus-test/functional-test/test_nydus.py:585]: Please trim trailing whitespace, perhaps try some kind of formater

File: contrib/nydus-test/functional-test/test_nydus.py [hunk 5]
```diff
@@ -747,10 +751,14 @@ def test_specified_prefetch(
     prefetching_files.append("/a/b/c/d")
     prefetching_files.append(os.path.join("/", "f/g/h/"))
 
-    specified_files = " ".join([os.path.join("/", d) for d in prefetching_files])
+    tmp_file = tmp_path / "prefetch.txt"
+    with open(tmp_file, "w") as f:
+        for d in prefetching_files:
+            f.write(os.path.join("/", d) + "\n")
```
Comment (changweige) [inline contrib/nydus-test/functional-test/test_nydus.py:757]: This can be simplified by using `writelines`

**Comment by changweige** [comment]
Otherwise, looks good to me

File: contrib/nydus-test/functional-test/test_nydus.py (hunk 3 above)
Comment (Desiki-high) [reply contrib/nydus-test/functional-test/test_nydus.py:567]: Yes,I run these test functions successfully . But I'm not familiar with pytest especially the fixture, so I'm not sure.

**Pull request merged**